### PR TITLE
Wos logging

### PR DESCRIPTION
**Pull request by atz**
Split off from #456 (without the filtering implementation).

### CODE REVIEW

File: lib/web_of_science.rb [hunk 1]
```diff
@@ -22,5 +22,8 @@ def self.queries
     @@queries ||= WebOfScience::Queries.new(client)
   end
 
+  def self.logger
+    @@logger ||= Logger.new(Settings.WOS.LOG)
+  end
```
Comment (darrenleeweber) [inline lib/web_of_science.rb]: This commit combines moving the namespace plus adding a new logger - these should be separate commits.  Split this commit into separate commits and then squash the addition of the logger with another commit that removes it from the NotificationManager, that commit is then a clear, atomic move-method refactor.

File: lib/notification_manager.rb [hunk 2]
```diff
@@ -40,10 +40,6 @@ def cap_logger
     def sciencewire_logger
       @@sciencewire_logger ||= Logger.new(Settings.SCIENCEWIRE.LOG)
     end
-
-    def wos_logger
-      @@wos_logger ||= Logger.new(Settings.WOS.LOG)
-    end
     # rubocop:enable Style/ClassVars
```
Comment (darrenleeweber) [inline lib/notification_manager.rb]: Merge this change with the addition of this method into the `WebOfScience` module, so it is clearer that this is a move-method refactor.

File: spec/lib/notification_manager_spec.rb [hunk 3]
```diff
@@ -66,17 +66,13 @@
...
-      expect(Logger).to receive(:new).with(Settings.WOS.LOG).once
-      described_class.error(exception, message, wos_client)
-    end
+    before { WebOfScience.class_variable_set(:@@logger, nil) }
+
     it 'logs errors' do
+      expect(WebOfScience).to receive(:logger).and_return(null_logger)
       expect(null_logger).to receive(:error).exactly(3)
       described_class.error(exception, message, wos_client)
     end
```
Comment (darrenleeweber) [inline spec/lib/notification_manager_spec.rb]: Replace
- `expect(WebOfScience).to receive(:logger).and_return(null_logger)`
- with `expect(Logger).to receive(:new).with(Settings.WOS.LOG).once.and_return(null_logger)`

Not sure why the `null_logger` is receiving 3 errors, is that right?

File: lib/web_of_science.rb (hunk 1 above)
Comment (atz) [reply lib/web_of_science.rb]: Why nitpick commit composition for changes you ostensibly want?  We are literally never going to use either state independently.  For this PR (and the previous one I already extracted these from), the **point** of the rename is to provide the logger method at the top `WebOfScience` module.  Breaking them apart gains nothing.

File: spec/lib/notification_manager_spec.rb (hunk 3 above)
Comment (atz) [reply spec/lib/notification_manager_spec.rb]: Nope.  That is 100% NOT the concern of `NotificationManager` anymore.

File: spec/lib/web_of_science/client_spec.rb [hunk 4]
```diff
@@ -61,18 +61,12 @@
...
+
+      it 'creates a logger and works' do
+        expect(WebOfScience).to receive(:logger).and_return(null_logger)
         expect(wos_client.session_close).to be_nil
       end
-      it 'creates a logger' do
-        expect(NotificationManager).to receive(:wos_logger).and_call_original
-        wos_client.session_close
-      end
     end
```
Comment (darrenleeweber) [inline spec/lib/web_of_science/client_spec.rb]: Best rspec pratices are to have only 1 expectation per spec.  This change is moving the code away from a best practice.  Apart from that, it is consistent with the intention to move the logger method (so that part of it is correct).

There is really only 1 line that needs to change in this spec.  At the previous line 66, the change would be to replace `NotificationManager` with `WebOfScience`.  That is the minimal change that works for the move-method refactor.

**Comment by atz** [comment]
@darrenleeweber on #456 and in person, you already said you were "OK, with the logging and module name change".  How is it you now have numerous objections to exactly the same code?

File: spec/lib/web_of_science/client_spec.rb (hunk 4 above)
Comment (atz) [reply spec/lib/web_of_science/client_spec.rb]: Read it again.  There is only one expectation.

File: spec/lib/notification_manager_spec.rb (hunk 3 above)
Comment (darrenleeweber) [reply spec/lib/notification_manager_spec.rb]: Huh?  I'm not talking about `NotificationManager`, right.

Comment (darrenleeweber) [reply spec/lib/notification_manager_spec.rb]: BTW, I think it gets 3 errors because there are 3 retries in the Faraday client.

File: spec/lib/web_of_science/client_spec.rb (hunk 4 above)
Comment (darrenleeweber) [reply spec/lib/web_of_science/client_spec.rb]: Huh?  The change collapses two specs into one, so it has two expectations, right?

**Comment by darrenleeweber** [comment]
These changes will not conflict with other PRs, if/when this is merged.

File: lib/web_of_science.rb (hunk 1 above)
Comment (darrenleeweber) [reply lib/web_of_science.rb]: I notice that renaming the module and moving the logger method are currently in two commits, but each commit contains a bit of both.  Either squash them all or separate the move module commit from the move method commit.

Comment (darrenleeweber) [reply lib/web_of_science.rb]: Yea, so just squash the first and last commits in this PR.

File: spec/lib/web_of_science_spec.rb [hunk 5]
```diff
@@ -26,4 +26,10 @@
       expect(result).to be_an WebOfScience::Queries
     end
   end
+
+  describe '#logger' do
+    it 'works' do
+      expect(described_class.logger).to be_a Logger
+    end
+  end
```
Comment (darrenleeweber) [inline spec/lib/web_of_science_spec.rb]: The better spec is to test that it uses the `Settings.WOS.LOG` value, so moving that spec into this spec file from the NotificationManager could resolve the questions on how that changed.

File: lib/web_of_science.rb (hunk 1 above)
Comment (atz) [reply lib/web_of_science.rb]: Squashed.

File: spec/lib/notification_manager_spec.rb (hunk 3 above)
Comment (atz) [reply spec/lib/notification_manager_spec.rb]: I think that is right.  In any case, the number of times is unchanged.

**Comment by darrenleeweber** [comment]
looks OK, but I want to hold off on a merge until a few other PRs are merged first.  This PR will require rebase for 3 or 4 open PRs because it changes the logger in all the WOS classes (not sure how many open PRs it will impact).  It will be much easier to rebase 476 than the other way around, the 476 rebase should be automatic and only needs to be done once after the other PRs are merged.

**Comment by darrenleeweber** [comment]
Ah, OK, looks like we have already merged a few other PRs and only a couple remain.  Going ahead with this.
